app/src/main/java/com/kinandcarta/create/proxytoggle/android/ThemeSwitcher.kt

```diff
@@ -0,0 +1,51 @@
+package com.kinandcarta.create.proxytoggle.android
+
+import android.content.Context
+import android.content.res.Configuration.UI_MODE_NIGHT_MASK
+import android.content.res.Configuration.UI_MODE_NIGHT_YES
+import androidx.appcompat.app.AppCompatDelegate
+import com.kinandcarta.create.proxytoggle.settings.AppSettings
+import dagger.hilt.android.qualifiers.ApplicationContext
+import dagger.hilt.android.scopes.ActivityScoped
+import javax.inject.Inject
+
+@ActivityScoped
+class ThemeSwitcher @Inject constructor(
+    @ApplicationContext context: Context,
```

Comment:
If it's Activity scoped could you use the ActivityContext here?

Reply:
You're absolutely right!

Reply:
Changed ✅